src/lsm/compaction.zig

```diff
@@ -198,11 +220,13 @@ pub fn CompactionType(
         }
 
         pub fn reset(compaction: *Compaction) void {
-            assert(compaction.state == .done);
+            assert(compaction.state == .done_applied_manifest);
+
             compaction.state = .idle;
+            compaction.manifest_entries.len = 0;
             if (compaction.grid_reservation) |grid_reservation| {
-                compaction.context.grid.forfeit(grid_reservation);
                 compaction.grid_reservation = null;
+                compaction.context.grid.forfeit(grid_reservation);
```

Comment:
Nulling out `grid_reservation` before passing it to `forfeit()`, and then relying on the capture to copy, was a surprising read. Would feel better to pass the memory directly, and then null.